packages/components/src/lib/d3/patchedScales.ts

```diff
@@ -0,0 +1,97 @@
+import * as d3 from "d3";
+
+type CustomFormat = "squiggle-default" | undefined;
+
+// see lib/d3/index.ts
+export const defaultTickFormatSpecifier: CustomFormat = "squiggle-default";
+
+function isCustomFormat(
+  specifier: string | undefined
+): specifier is CustomFormat {
+  return specifier === "squiggle-default" || specifier === undefined;
+}
+
+function squiggleDefaultFormat() {
+  const locale = d3.formatLocale({
+    decimal: ".",
+    thousands: ",",
+    grouping: [3],
+    currency: ["$", ""],
+    minus: "-",
+  });
+  const siFormat = locale.format(".9~s");
+  const expFormat = locale.format(".9~e");
+  const fixedFormat = locale.format(".9~f");
+
+  return (d: d3.NumberValue) => {
+    const abs = Math.abs(Number(d));
+    if (abs === 0 || (abs >= 0.0001 && abs < 1e6)) {
```

Comment:
I suggest allowing 0.00001 as well. I'd expect people to have an easier time parsing that than 10e-5. (That said, this might be too long for the UI, in which case we shouldn't do it)